src/Sulu/Bundle/SecurityBundle/Tests/Unit/SingleSignOn/Adapter/OpenId/OpenIdSingleSignOnAdapterTest.php

```diff
@@ -180,20 +184,54 @@ function($method, $url, $options): MockResponse {
             ->willReturn('https://sulu.io/admin');
 
         $this->userRepository->findOneBy(Argument::any())->willReturn(null);
-        $this->contactRepository->createNew()->willReturn($this->prophesize(Contact::class)->reveal());
-        $this->entityManager->persist(Argument::any())->shouldBeCalled();
-        $role = $this->prophesize(Role::class);
-        $role->getAnonymous()->shouldBeCalled()->willReturn(false);
-        $role->getIdentifier()->willReturn('hello@sulu.io');
+
+        $user = new User();
+        $this->userRepository->createNew()->willReturn($user)->shouldBeCalled();
+
+        $contact = new Contact();
+        $this->contactRepository->createNew()->willReturn($contact)->shouldBeCalled();
+
+        $persistedUser = null;
+        $persistedContact = null;
+        $persistedUserRole = null;
+
+        $this->entityManager->persist(Argument::that(function($object) use (&$persistedUser, &$persistedContact, &$persistedUserRole) {
+            if ($object instanceof Contact) {
+                $persistedContact = $object;
+            } elseif ($object instanceof User) {
+                $persistedUser = $object;
+            } elseif ($object instanceof UserRole) {
+                $persistedUserRole = $object;
+            } else {
+                $this->fail('Unexpected object: ' . $object::class);
+            }
+
+            return true;
+        }))->shouldBeCalledTimes(3);
+        $role = new Role();
+        $role->setKey('ADMIN');
         $this->roleRepository->findOneBy(Argument::any())
             ->shouldBeCalled()
-            ->willReturn($role->reveal());
+            ->willReturn($role);
         $this->entityManager->flush()->shouldBeCalled();
 
-        $expectedUserBadge = new UserBadge('hello@sulu.io', null, ['email' => 'hello@sulu.io']);
+        $expectedUserBadge = new UserBadge('hello@sulu.io', null, [
+            'email' => 'hello@sulu.io',
+            'family_name' => 'Sulu',
+            'given_name' => 'Hikaru',
+        ]);
 
         $result = $this->adapter->createOrUpdateUser($token);
 
         $this->assertEquals($expectedUserBadge, $result);
+
+        $this->assertNotNull($persistedUser);
+        $this->assertSame($persistedUser, $persistedUserRole->getUser());
+        $this->assertNotNull($persistedContact);
+        $this->assertSame('Hikaru', $persistedContact->getFirstName());
+        $this->assertSame('Sulu', $persistedContact->getLastName());
+        $this->assertSame('hello@sulu.io', $persistedUser->getEmail());
+        $this->assertNotNull($persistedUserRole);
+        $this->assertSame($role, $persistedUserRole->getRole());
```

Comment:
this was not yet tested so I added tests to check what we did persist

Reply:
Thank you!